src/partition/repart.c

```diff
@@ -5278,7 +5279,20 @@ static int context_minimize(Context *context) {
                 /* Read-only filesystems are minimal from the first try because they create and size the
                  * loopback file for us. */
                 if (fstype_is_ro(p->format)) {
+                        fd = open(temp, O_RDONLY|O_CLOEXEC);
+                        if (fd < 0)
+                                return log_error_errno(errno, "Failed to open temporary file %s: %m", temp);
+
```

Comment:
as extra paranoia you might do `fd_verify_regular()` here and fail if it fails? it's usually a good idea before starting to use stuff